### PR TITLE
ci: don't run code tests in documentation changes

### PR DESCRIPTION
**Pull request by FFY00**

### CODE REVIEW

**Comment by layday** [comment]
Good thinking, should probably add the changelog and readme to ignores as well.

File: .github/workflows/test.yml
Comment (gaborbernat) [inline .github/workflows/test.yml:109]: This env absolutely must run on documentation changes.

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml
Comment (webknjaz) [inline .github/workflows/test.yml:94]:
```suggestion
```

File: .github/workflows/test.yml
Comment (webknjaz) [reply .github/workflows/test.yml:109]: FWIW RTD integration runs on PRs so it's roughly equivalent. I'd suggest adding a strict mode to avoid broken references completely, though...

File: .github/workflows/test.yml
Comment (webknjaz) [inline .github/workflows/test.yml:111]: It's cleaner to just set an env var instead of passing `-e` to each step... Plus here's a few extra args that make tox output nicer in GHA in some cases:
```suggestion
    env:
      PY_COLORS: 1
      TOXENV: ${{ matrix.tox_env }}
      TOX_PARALLEL_NO_SPINNER: 1

    steps:
```

Comment (FFY00) [reply .github/workflows/test.yml:104]: I applied these changes, with the exception of `TOXENV`, as we no longer have a matrix for tox jobs, they are run in different workflows. Will keep in mind though, thanks!

File: .github/workflows/test.yml
Comment (FFY00) [reply .github/workflows/test.yml]: Ups, I have put the docs run in the check workflow.

**Comment by henryiii** [comment]
I think the required checks in the settings -> branches -> main branch protection rules need to be updated right before this is merged. The check / docs and check / type names has changed.

**Comment by FFY00** [comment]
Yes, I am just waiting for approvals. I think I have addressed everything that was pointed out.

**Review by layday — APPROVED** [review]

**Pull request merged**
